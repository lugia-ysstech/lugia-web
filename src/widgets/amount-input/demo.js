/**
 *
 * create by ligx
 *
 * @flow
 */
import * as React from 'react';
import Theme from '../theme';
import AmountInput from './index';
import Widget from '../consts/index';
import styled from 'styled-components';
import { getBoxShadow } from '@lugia/theme-utils';

class LimitAmountInput extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = { value: '' };
  }

  onChange = ({ newValue: value }: any) => {
    this.setState({ value });
    this.props.onChange({ newValue: value });
  };

  render() {
    return <AmountInput value={this.state.value} onChange={this.onChange} />;
  }
}
class DefaultValueAmountInput extends React.Component<any, any> {
  render() {
    return <AmountInput defaultValue={'123456'} onChange={this.props.onChange} />;
  }
}
const Wrapper = styled.div`
  float: left;
  margin-left: 50px;
`;
export default () => {
  const view = {
    [Widget.AmountInput]: {
      Container: {
        normal: {
          width: 400,
          height: 40,
          border: {
            top: 10,
          },
        },
      },
      AmountInputPrefix: { normal: { fontSize: 14, color: 'blue' } },
      AmountTip: {
        Container: {
          normal: {
            background: {
              color: '#eee',
            },
          },
        },
        TooltipTitle: {
          normal: {
            color: '#4d63ff',
            fontSize: 16,
          },
        },
      },
    },
  };
  const onChange = (cmpName: string) => (value: any) => {};
  return (
    <div>
      <Wrapper>
        <p>formatter AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>default AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>禁用状态 </p>
        <AmountInput size={'default'} disabled={true} />
      </Wrapper>
      <Wrapper>
        <Theme config={view}>
          <p>small size</p>
          <AmountInput size={'small'} placeholder={'请填写金额'} />
          <p>default size</p>
          <AmountInput placeholder={'请填写金额'} />
          <p>large size</p>
          <AmountInput size={'large'} placeholder={'请填写金额'} disabled={true} />
        </Theme>
      </Wrapper>
      <Wrapper>
        <p>受限Input</p>
        <LimitAmountInput onChange={onChange('limit')} />
        <p>有默认值的 受限Input</p>
        <DefaultValueAmountInput value={'123456'} onChange={onChange('limit')} />
      </Wrapper>
      <Wrapper>
        <p>amountPrefix: '¥' transform: false </p>
        <AmountInput amountPrefix="¥" transform={false} />
        <p>amountPrefix: '$' transform: false </p>
        <AmountInput amountPrefix="$" transform={false} />
      </Wrapper>
    </div>
  );
};
