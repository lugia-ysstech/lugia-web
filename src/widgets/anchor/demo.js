/**
 *
 * create by guorg
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';
import Theme from '../theme';
import Widget from '../consts/index';
import Anchor from './index';

const { Link } = Anchor;
const DemoBox = styled.div`
  padding: 20px;
  margin-left: ${props => 120 + props.cur * 120}px;
  width: 209px;
  height: 203px;
`;

export default class DrawerDemo extends React.Component<any, any> {
  constructor() {
    super();
  }
  click = () => {
    const anchorElement = document.getElementById('anchor-link');
    if (anchorElement) {
      anchorElement.scrollIntoView();
    }
  };

  handleLinkClick = (e, href) => {
    if (href) {
      const name = href.slice(1);
      const anchorElement = document.getElementById(name);
      if (anchorElement) {
        anchorElement.scrollIntoView({ block: 'start', behavior: 'smooth' });
      }
    }
  };
  render() {
    return (
      <div>
        <DemoBox cur={0}>
          <Anchor slideType="circle" useHref={false} onClick={this.handleLinkClick}>
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <DemoBox cur={1}>
          <Anchor slideType="line">
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <DemoBox cur={2}>
          <Anchor slideType="none" affix={false}>
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <div style={{ marginTop: '100px' }} id="anchor-link1" name="anchor-link1">
          anchor-link1
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link2" name="anchor-link2">
          anchor-link2
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link3" name="anchor-link3">
          anchor-link3
        </div>
        <div style={{ marginTop: '200px' }} id="anchor-link3.1" name="anchor-link3.1">
          anchor-link3.1
        </div>
        <div style={{ marginTop: '200px' }} id="anchor-link3.2" name="anchor-link3.2">
          anchor-link3.2
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link4" name="anchor-link4">
          anchor-link4
        </div>
      </div>
    );
  }
}
