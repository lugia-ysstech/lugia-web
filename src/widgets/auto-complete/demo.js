import * as React from 'react';
import AutoComplete from './';
import { getBorder } from '@lugia/theme-utils';
import Widget from '../consts/index';
import { getBorderRadius } from '../theme/CSSProvider';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const config = {
      [Widget.AutoComplete]: {
        OldItem: {
          normal: {
            height: 50,
            color: '#000',
            font: { size: 20 },
            background: { color: '#333' },
            opacity: 0.6,
            padding: { left: 10 },
            border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(20),
          },
          hover: {
            color: '#4d63ff',
            background: { color: '#4a8e29' },
            opacity: 1,
            border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(20),
          },
        },

        Container: {
          normal: {
            width: 300,
            height: 60,
          },
        },

        Menu: {
          MenuWrap: {
            normal: {
              width: 300,
              height: 350,
              opacity: 0.6,
              boxShadow: '2px 2px 5px #4d63ff',
              background: { color: '#000' },
              border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
              borderRadius: getBorderRadius(20),
            },
            hover: {
              opacity: 1,
            },
          },
          MenuItem: {
            normal: { color: '#ccc', fontSize: 14, font: { fontWeight: 900 } },
            hover: {
              color: '#fff',
              fontSize: 20,
              background: { color: 'green' },
              font: { fontWeight: 400 },
            },
            active: {
              color: 'blue',
              fontSize: 14,
              background: { color: 'pink' },
              font: { fontWeight: 900 },
            },
            disabled: { color: 'red', background: { color: '#000' } },
          },
          SelectedMenuItem: {
            normal: {
              color: 'blue',
              font: { fontWeight: 900 },
              fontSize: 18,
              background: { color: 'orange' },
            },
            hover: { color: '#000', background: { color: 'yellow' } },
            active: { color: 'green' },
          },
          Divider: { normal: { color: 'red' } },
        },
      },
    };
    const { menuData, value } = this.state;
    return (
      <AutoComplete
        theme={config}
        showOldValue
        value={value}
        data={menuData}
        onChange={this.onChange}
      />
    );
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
