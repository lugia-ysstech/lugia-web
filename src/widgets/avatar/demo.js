/**
 *
 * create by liangguodong on 2018/8/27
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Avatar from './';
import Widget from '../consts';
import Theme from '../theme';
import Badge from '../badge/index';

const Wrapper = styled.div`
  float: left;
  margin-left: 50px;
  text-align: center;
  padding: 20px;
`;
export default () => {
  const badgeConfig = {
    [Widget.Badge]: {
      BadgeDot: {
        normal: {
          position: { right: 0, top: -3 },
        },
      },
      BadgeNumber: {
        InnerNumber: {
          normal: { position: { right: -5, top: -3 } },
        },
      },
    },
  };
  const config = {
    [Widget.Avatar]: {
      Container: { normal: { background: { color: '#e2e2e2' } } },
      IconAvatar: { normal: { color: 'orange' } },
      SrcAvatar: { normal: { width: 24, height: 24 } },
      FontAvatar: { normal: { color: 'red' } },
    },
  };
  const iconConfig = {
    [Widget.Avatar]: {
      IconAvatar: { normal: { color: 'purple', fontSize: 16 } },
    },
  };
  return (
    <div>
      <Theme config={config}>
        <Wrapper>
          <Avatar shape={'square'} name={'l'} />
          <Avatar shape={'square'} name={'lu'} />
          <Avatar viewClass="config" shape={'square'} name={'lug'} />
          <Avatar viewClass="config" shape={'square'} name={'lugia'} />
          <br />
          <Avatar shape={'circle'} name={'l'} />
          <Avatar shape={'circle'} name={'lu'} />
          <Avatar viewClass="config" shape={'circle'} name={'lug'} />
          <Avatar viewClass="config" shape={'circle'} name={'lugia'} />
        </Wrapper>
        <Wrapper>
          <Theme config={iconConfig}>
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'small'}
            />
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'default'}
            />
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'large'}
            />
          </Theme>
          <br />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} size={'small'} />
          <Avatar
            shape={'circle'}
            type="icon"
            icon={'lugia-icon-financial_user'}
            size={'default'}
          />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} size={'large'} />
        </Wrapper>
        <Wrapper>
          <Theme
            config={{
              register: {
                FontAvatar: { normal: { color: 'blue' } },
              },
            }}
          >
            <Avatar viewClass="register" shape={'square'} size={'small'} name={'lugia'} />
            <Avatar viewClass="register" shape={'square'} size={'default'} name={'lugia'} />
            <Avatar viewClass="register" shape={'square'} size={'large'} name={'lugia'} />
          </Theme>
          <br />
          <Avatar shape={'circle'} size={'small'} name={'lugia'} />
          <Avatar shape={'circle'} size={'default'} name={'lugia'} />
          <Avatar shape={'circle'} size={'large'} name={'lugia'} />
        </Wrapper>
        <Wrapper>
          <Avatar shape={'square'} name={'lugia'} msgNum={1} />
          <Avatar shape={'square'} type="icon" icon={'lugia-icon-financial_user'} msgNum={2} />
          <Avatar
            shape={'square'}
            type="img"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/master/lugiaweb%E7%BB%84%E4%BB%B6/%E5%A4%B4%E5%83%8F/32.jpg"
            msgNum={99}
          />
          <br />
          <Avatar shape={'circle'} name={'lugia'} />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} />
          <Avatar
            shape={'circle'}
            type="img"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/master/lugiaweb%E7%BB%84%E4%BB%B6/%E5%A4%B4%E5%83%8F/32.jpg"
            msgNum={99}
          />
        </Wrapper>
      </Theme>

      <Wrapper>
        <Theme config={badgeConfig}>
          <Badge count={10}>
            <Avatar shape={'square'} size={'large'} name={'lugia'} />
          </Badge>
          <br />
          <br />
          <Badge dot={true}>
            <Avatar shape={'square'} size={'large'} name={'lugia'} />
          </Badge>
        </Theme>
        <br />
        <br />
      </Wrapper>
      <Wrapper>
        <Theme config={badgeConfig}>
          <Badge viewClass="badgeConfig" count={10}>
            <Avatar icon={'lugia-icon-financial_user'} size={'large'} type="icon" />
          </Badge>
          <br />
          <br />
          <Badge viewClass="badgeConfig" dot={true}>
            <Avatar icon={'lugia-icon-financial_user'} size={'large'} type="icon" />
          </Badge>
        </Theme>
      </Wrapper>
    </div>
  );
};
