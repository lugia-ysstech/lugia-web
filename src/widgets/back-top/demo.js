/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';
import { getBorder } from '@lugia/theme-utils';
import { getBorderRadius } from '../theme/CSSProvider';

const Demo = styled.div`
  & > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        Container: {
          normal: {
            color: 'red',
            background: { color: 'orange' },
            width: 100,
            height: 100,
            opacity: 0.7,
            border: getBorder({ color: 'red', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(100),
          },
        },
        BackTopIcon: {
          normal: {
            color: 'green',
            fontSize: 16,
            padding: {
              right: 10,
              left: 10,
              top: 10,
              bottom: 10,
            },
            margin: {
              right: 0,
              left: 0,
              top: 0,
              bottom: 0,
            },
          },
        },
      },
    };

    return (
      <div style={{ height: '1600px' }}>
        <Demo cur={1}>
          <Theme>
            <BackTop showType="textType" text="up" />
          </Theme>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop icon={'lugia-icon-logo_apple'} showType="iconType" />
          </Theme>
        </Demo>
        <Demo>
          <Theme>
            <BackTop />
          </Theme>
        </Demo>
        <Demo cur={4}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={4}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop
              showType="textType"
              text="up"
              visibilityHeight={80}
              target={() => this.EleRef}
            />
          </Demo>
        </div>
        <br />
        <br />
        <h2>injectLugiad属性, 展示使用</h2>
        <div style={{ margin: '20px' }}>
          <BackTop showType="textType" text="up" injectLugiad={{ type: 'BackTop' }} />
        </div>
      </div>
    );
  }
}
