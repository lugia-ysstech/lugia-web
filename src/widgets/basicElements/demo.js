import React, { Component } from 'react';
import BasicElements from './';
import Widgets from '../consts/index';
import Theme from '../theme/index';
import { getBorderRadius, getBorder, getBoxShadow } from '@lugia/theme-utils';
class Demo extends Component {
  render() {
    return (
      <div>
        <Theme
          config={{
            [Widgets.BasicElements]: {
              Container: {
                normal: {
                  width: 200,
                  height: 300,
                  background: {
                    color: 'green',
                  },
                  borderRadius: getBorderRadius(20),
                  border: getBorder({ color: 'blue', width: 1, style: 'dashed' }),
                  boxShadow: getBoxShadow('2 2 20 red'),
                  margin: { top: 20, right: 20, bottom: 20, left: 20 },
                  opacity: 0.6,
                },
              },
            },
          }}
        >
          <BasicElements />
          <BasicElements shape={'circular'} />
          <BasicElements shape={'triangle'} />
        </Theme>
      </div>
    );
  }
}

export default Demo;
