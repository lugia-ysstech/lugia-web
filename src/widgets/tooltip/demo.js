/**
 *
 * create by ligx
 *
 * @flow
 */
import * as React from 'react';
import Tooltip from './';
import Button from '../button';
import Widget from '../consts/index';
import Theme from '../theme';
import styled from 'styled-components';

const buttonWidth = 80;
const text = 'prompt text';
const ButtonDemo = styled(Button)`
  width: ${buttonWidth}px;
`;
const ToolTipBaseWrapper = styled.div`
  display: inline-block;
`;
const OuterWrapper = styled.div`
  margin-left: 100px;
`;
const TestContainer = styled.div`
  width: 200px;
  position: relative;
  height: 200px;
  border: 1px red solid;
  margin-left: 100px;
`;
const TestDiv = styled.div`
  width: 50px;
  height: 20px;
  border: 1px blue solid;
`;
TestDiv.displayName = 'TestDiv';

const DirectionTopWrapper = styled.div`
  margin-left: 120px;
  white-space: nowrap;
  display: inline-block;
`;
const DirectionLeftWrapper = styled.div`
  width: ${buttonWidth}px;
  position: absolute;
`;
const DirectionRightWrapper = styled.div`
  width: ${buttonWidth}px;
  margin-left: ${buttonWidth * 4 + 64}px;
`;
const DirectionBottomWrapper = styled.div`
  margin-left: 120px;
  white-space: nowrap;
  display: inline-block;
`;
const BottomWrapper = styled.div`
  margin-top: 100px;
`;
const ToolTipBHWrapper = styled(ToolTipBaseWrapper)`
  margin-bottom: 50px;
  margin-left: 10px;
  margin-top: 10px;
`;
const ToolTipTHWrapper = styled(ToolTipBaseWrapper)`
  margin-top: 50px;
  margin-left: 10px;
`;
const ToolTipLVWrapper = styled(ToolTipBaseWrapper)`
  margin-top: 10px;
  margin-left: 50px;
`;
const ToolTipRVWrapper = styled(ToolTipBaseWrapper)`
  margin-top: 10px;
  margin-right: 50px;
  margin-left: 10px;
`;

const childrenContainerConfig = {
  [Widget.Tooltip]: {
    ChildrenContainer: {
      normal: {
        width: '100%',
      },
    },
  },
};
const containerConfig = {
  [Widget.Tooltip]: {
    Container: {
      normal: {
        width: 120,
        height: 50,
      },
    },
  },
};

export const Demo = () => {
  const config = {
    [Widget.Button]: {
      Container: {
        normal: {
          width: buttonWidth,
        },
      },
    },
  };
  return (
    <OuterWrapper>
      <Theme config={config}>
        <DirectionTopWrapper>
          <ToolTipTHWrapper>
            <Tooltip placement="topLeft" title={text}>
              <ButtonDemo type="primary">TL</ButtonDemo>
            </Tooltip>
          </ToolTipTHWrapper>
          <ToolTipTHWrapper>
            <Tooltip placement="top" title={text}>
              <ButtonDemo type="primary">Top</ButtonDemo>
            </Tooltip>
          </ToolTipTHWrapper>
          <ToolTipTHWrapper>
            <Tooltip placement="topRight" title={text}>
              <ButtonDemo type="primary">TR</ButtonDemo>
            </Tooltip>
          </ToolTipTHWrapper>
        </DirectionTopWrapper>
        <DirectionLeftWrapper>
          <ToolTipLVWrapper>
            <Tooltip placement="leftTop" title={text}>
              <ButtonDemo type="primary">LT</ButtonDemo>
            </Tooltip>
          </ToolTipLVWrapper>
          <ToolTipLVWrapper>
            <Tooltip placement="left" title={text}>
              <ButtonDemo type="primary">Left</ButtonDemo>
            </Tooltip>
          </ToolTipLVWrapper>
          <ToolTipLVWrapper>
            <Tooltip placement="leftBottom" title={text}>
              <ButtonDemo type="primary">LB</ButtonDemo>
            </Tooltip>
          </ToolTipLVWrapper>
        </DirectionLeftWrapper>
        <DirectionRightWrapper>
          <ToolTipRVWrapper>
            <Tooltip placement="rightTop" title={text}>
              <ButtonDemo type="primary">RT</ButtonDemo>
            </Tooltip>
          </ToolTipRVWrapper>
          <ToolTipRVWrapper>
            <Tooltip placement="right" title={text}>
              <ButtonDemo type="primary">Right</ButtonDemo>
            </Tooltip>
          </ToolTipRVWrapper>
          <ToolTipRVWrapper>
            <Tooltip placement="rightBottom" title={text}>
              <ButtonDemo type="primary">RB</ButtonDemo>
            </Tooltip>
          </ToolTipRVWrapper>
        </DirectionRightWrapper>
        <DirectionBottomWrapper>
          <ToolTipBHWrapper>
            <Tooltip placement="bottomLeft" title={text}>
              <ButtonDemo type="primary">BL</ButtonDemo>
            </Tooltip>
          </ToolTipBHWrapper>
          <ToolTipBHWrapper>
            <Tooltip placement="bottom" title={text}>
              <ButtonDemo type="primary">Bottom</ButtonDemo>
            </Tooltip>
          </ToolTipBHWrapper>
          <ToolTipBHWrapper>
            <Tooltip placement="bottomRight" title={text}>
              <ButtonDemo type="primary">BR</ButtonDemo>
            </Tooltip>
          </ToolTipBHWrapper>
        </DirectionBottomWrapper>
        <BottomWrapper />
      </Theme>
      <div>
        <Tooltip placement="top" title={text} description={'This is description'}>
          <ButtonDemo type="primary">带有描述文本的提示</ButtonDemo>
        </Tooltip>
      </div>
      <p>包含的组件 继承容器的高度</p>
      <div style={{ width: 100, height: 50 }}>
        <Tooltip title={text} action={'click'} placement="top">
          <div style={{ height: '100%', border: '1px solid red' }}>点击触发</div>
        </Tooltip>
      </div>
      <p>包含的组件 继承容器的宽度 </p>
      <div style={{ width: 100, height: 50 }}>
        <Theme config={childrenContainerConfig}>
          <Tooltip title={text} action={'click'} placement="top">
            <div style={{ height: '100%', border: '1px solid red' }}>点击触发</div>
          </Tooltip>
        </Theme>
      </div>
      <div style={{ width: 100, height: 50 }}>
        <Theme config={containerConfig}>
          <Tooltip
            title={'This is an English title that wants to wrap a line'}
            action={'click'}
            placement="top"
          >
            <ButtonDemo type="primary">英文 可换行的title</ButtonDemo>
          </Tooltip>
        </Theme>
      </div>
    </OuterWrapper>
  );
};
export const TestDemo = () => {
  return (
    <TestContainer>
      <p>距左边距离不够弹窗时 在右边弹窗 箭头位于左边 </p>
      <div>
        <Tooltip
          placement={'left'}
          title={<div style={{ width: 100, height: 50 }}> this is title</div>}
        >
          <TestDiv>test</TestDiv>
        </Tooltip>
      </div>
    </TestContainer>
  );
};

export default () => {
  return [<Demo />, <TestDemo />];
};
