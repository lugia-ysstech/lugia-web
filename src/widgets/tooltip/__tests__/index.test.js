//@flow
import React from 'react';
import chai from 'chai';
import 'jest-styled-components';
import Tooltip from '../';
import { TestDemo } from '../demo';
import { createTestComponent } from '@lugia/react-test-utils';

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const { expect: exp } = chai;

describe('Tooltip', () => {
  it('getDirection', () => {
    const Target = createTestComponent(Tooltip, target => {
      exp(target.getThemeTarget().getDirection('left')).to.be.equal('right');
      exp(target.getThemeTarget().getDirection('leftTop')).to.be.equal('right');
      exp(target.getThemeTarget().getDirection('leftBottom')).to.be.equal('right');

      exp(target.getThemeTarget().getDirection('right')).to.be.equal('left');
      exp(target.getThemeTarget().getDirection('rightTop')).to.be.equal('left');
      exp(target.getThemeTarget().getDirection('rightBottom')).to.be.equal('left');

      exp(target.getThemeTarget().getDirection('top')).to.be.equal('bottom');
      exp(target.getThemeTarget().getDirection('topLeft')).to.be.equal('bottom');
      exp(target.getThemeTarget().getDirection('topRight')).to.be.equal('bottom');

      exp(target.getThemeTarget().getDirection('bottom')).to.be.equal('top');
      exp(target.getThemeTarget().getDirection('bottomLeft')).to.be.equal('top');
      exp(target.getThemeTarget().getDirection('bottomRight')).to.be.equal('top');
    });
    mount(
      <Target>
        <button />
      </Target>
    );
  });

  it('Snapshot JSON', () => {
    global.svtest = true;
    const component = mount(<TestDemo />);
    ReactDOM.createPortal = node => node;
    component.find('TestDiv').simulate('click');
    expect(renderer.create(component).toJSON()).toMatchSnapshot();
    global.svtest = false;
  });
});
