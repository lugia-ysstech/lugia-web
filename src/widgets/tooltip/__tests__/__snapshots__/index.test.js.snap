// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Snapshot JSON 1`] = `
.c3 {
  display: none;
}

.c4 {
  position: absolute;
  border-radius: 4px;
  box-sizing: border-box;
  left: -9999px;
  top: -9999px;
  z-index: undefined;
}

.c5 {
  position: relative;
  padding-right: 0.4rem;
  background: transparent;
  box-shadow: none;
}

.c6 {
  position: relative;
  box-sizing: border-box;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
}

.c7 {
  border-color: transparent;
  position: absolute;
  border-style: solid;
  line-height: 1;
  top: calc(50% - 5px);
  right: -0.3rem;
  border-left-color: #363B40;
  border-width: 0.5rem 0 0.5rem 0.5rem;
}

.c8 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  font-size: 1.2rem;
  color: #ffffff;
}

.c1 {
  display: inline-block;
  height: 100%;
}

.c0 {
  width: 200px;
  position: relative;
  height: 200px;
  border: 1px red solid;
  margin-left: 100px;
}

.c2 {
  width: 50px;
  height: 20px;
  border: 1px blue solid;
}

<div
  className="c0"
>
  <p>
    距左边距离不够弹窗时 在右边弹窗 箭头位于左边 
  </p>
  <div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <div
        className="c2"
      >
        test
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c5"
          style={Object {}}
        >
          <div
            className="c6"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "#363B40",
                "borderBottomLeftRadius": "0.4rem",
                "borderBottomRightRadius": "0.4rem",
                "borderTopLeftRadius": "0.4rem",
                "borderTopRightRadius": "0.4rem",
                "boxShadow": "0px 0px 6px 0px #d0d3da",
              }
            }
          >
            <div
              className="c7"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
            <div
              className="c8"
              style={Object {}}
            >
              <div
                style={
                  Object {
                    "height": 50,
                    "width": 100,
                  }
                }
              >
                 this is title
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
